src/components/MessageInput/hooks/useSubmitHandler.ts

```diff
@@ -42,37 +43,53 @@ export const useSubmitHandler = <
   );
   const { t } = useTranslationContext('useSubmitHandler');
 
+  const textReference = useRef({ hasChanged: false, initialText: text });
+
+  useEffect(() => {
+    if (!textReference.current.initialText.length) {
+      textReference.current.initialText = text;
+      return;
+    }
+
+    textReference.current.hasChanged = text !== textReference.current.initialText;
+  }, [text]);
+
   const getAttachmentsFromUploads = () => {
     const imageAttachments = imageOrder
       .map((id) => imageUploads[id])
       .filter((upload) => upload.state !== 'failed')
       .filter((
         { id, url },
         _,
-        self, // filter out duplicates based on url
+        self, // filter out duplicates based on ID or URL
       ) => self.every((upload) => upload.id === id || upload.url !== url))
-      .map(
-        (upload) =>
-          ({
-            fallback: upload.file.name,
-            image_url: upload.url,
-            type: 'image',
-          } as Attachment<StreamChatGenerics>),
-      );
+      .filter((upload) => {
```

Comment:
@arnautov-anton probably needs unit tests to check that the logic of `hasChanged` is covered?